core/services/workflows/syncer/handler.go

```diff
@@ -170,32 +311,47 @@ func (h *eventHandler) workflowActivatedEvent(
 // forceUpdateSecretsEvent handles the ForceUpdateSecretsEvent event type.
 func (h *eventHandler) forceUpdateSecretsEvent(
 	ctx context.Context,
-	event WorkflowRegistryEvent,
+	payload WorkflowRegistryForceUpdateSecretsRequestedV1,
 ) error {
 	// Get the URL of the secrets file from the event data
-	data, ok := event.Data.(WorkflowRegistryForceUpdateSecretsRequestedV1)
-	if !ok {
-		return fmt.Errorf("invalid data type %T for event", event.Data)
-	}
-
-	hash := hex.EncodeToString(data.SecretsURLHash)
+	hash := hex.EncodeToString(payload.SecretsURLHash)
 
 	url, err := h.orm.GetSecretsURLByHash(ctx, hash)
 	if err != nil {
-		h.lggr.Errorf("failed to get URL by hash %s : %s", hash, err)
-		return err
+		return fmt.Errorf("failed to get URL by hash %s : %w", hash, err)
 	}
 
 	// Fetch the contents of the secrets file from the url via the fetcher
 	secrets, err := h.fetcher(ctx, url)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to fetch secrets from url %s : %w", url, err)
 	}
 
 	// Update the secrets in the ORM
 	if _, err := h.orm.Update(ctx, hash, string(secrets)); err != nil {
-		return err
+		return fmt.Errorf("failed to update secrets: %w", err)
 	}
 
 	return nil
 }
+
+// sha calculates the sha256 hash of the wasm, config and secretsURL to determine the workflow ID.
+func sha(wasm, config, secretsURL []byte) string {
+	sum := sha256.New()
+	sum.Write(wasm)
+	sum.Write(config)
+	sum.Write(secretsURL)
+	return hex.EncodeToString(sum.Sum(nil))
+}
+
+// logCustMsg emits a custom message to the external sink and logs an error if that fails.
+func logCustMsg(ctx context.Context, cma custmsg.MessageEmitter, msg string, log logger.Logger) {
+	err := cma.Emit(ctx, msg)
+	if err != nil {
+		log.Errorf("failed to send custom message with msg: %s, err: %v", msg, err)
```

Comment:
Will it be problematic that this log line will point here rather than the caller?

Reply:
hm, potentially, the context will be in the message, but do you suggest I just return the error instead and let the caller handle it?

Reply:
If handling in the caller is annoying, then you can just do:
```suggestion
		log.Helper(1).Errorf("failed to send custom message with msg: %s, err: %v", msg, err)
```
But it constructs a new logger, so not great if it's on the hot path. nbd if this is truly exceptional though.